### ISSUE TITLE
Rethink fraction implementations of Staking

### ISSUE THREAD

**Issue by Kianenigma** [paritytech/substrate#2558]
- Once https://github.com/paritytech/substrate/pull/2501 is merged Per-things are safer to use and can replace the manual `ACCURACY * thing / thing`.

- If possible, extend this approach to `PerU128` as well (or eliminate it overall if possible)

- piggy-back: double check the stake-loss-due-to-rounding-regain loop.

**Comment by Kianenigma** [comment]
More or less a duplicate of https://github.com/paritytech/substrate/issues/2908
